### PR TITLE
Support publishing generator

### PR DESCRIPTION
**Pull request by OlegKipchatov**

### CODE REVIEW

File: .github/actions/detect-version-generator-bump/action.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Detect Version Bump
+outputs:
+  changed:
+    description: true if version is bumped
```
Comment (AnnaKondratova) [inline .github/actions/detect-version-generator-bump/action.yml:4]:
```suggestion
    description: 'true' if the generator version is bumped
```

File: .github/actions/detect-version-generator-bump/action.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    value: ${{ steps.detection.outputs.changed }}
+  version:
+    description: new version
```
Comment (AnnaKondratova) [inline .github/actions/detect-version-generator-bump/action.yml:7]:
```suggestion
    description: New version
```

File: .github/actions/detect-version-generator-bump/action.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  using: composite
+  steps:
+    - shell: bash
+      id: detection
+      run: |
+        versionDiff=$( git diff HEAD^ -U0 -- packages/devextreme-angular-generator/package.json )
+        if ! [[ $versionDiff ]] || ! [[ $versionDiff =~ (\+ *\"version\":) ]]
+        then
+          echo "::set-output name=changed::false"
+          echo "Version generator didn't change"
```
Comment (AnnaKondratova) [inline .github/actions/detect-version-generator-bump/action.yml:19]:
```suggestion
          echo "The generator version is not changed"
```

File: .github/actions/detect-version-generator-bump/action.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          exit 0
+        fi
+
+        newVersion=$(echo $versionDiff | grep -oP "\+ *\"version\":.+," | grep -oP "\d+\.\d+.\d+(-beta\.\d+)?")
+        echo "Version generator changed to $newVersion"
```
Comment (AnnaKondratova) [inline .github/actions/detect-version-generator-bump/action.yml:24]:
```suggestion
        echo "The generator version changed to $newVersion"
```

File: .github/workflows/version-generator.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+name: Version Generator
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:1]:
```suggestion
name: Generator Version
```

File: .github/workflows/version-generator.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+on:
+  pull_request:
+    paths:
+      - packages/devextreme-angular-generator/**
+    types:
+      - opened
+      - reopened
+      - synchronize
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:10]:
```suggestion
      - synchronized
```

File: .github/workflows/version-generator.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    runs-on: ubuntu-latest
+    name: Bump
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Detect files generator changed
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:24]:
```suggestion
      - name: Detect changes in the generator files
```

File: .github/workflows/version-generator.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        id: filesChange
+        uses: ./.github/actions/detect-generator-changes
+
+      - name: Detect version generator bump
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:28]:
```suggestion
      - name: Detect the generator version bump
```

File: .github/workflows/version-generator.yml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        id: versionBump
+        uses: ./.github/actions/detect-version-generator-bump
+
+      - name: Check version generator bump PR
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:32]:
```suggestion
      - name: Check the generator version bump (PR)
```

File: .github/workflows/version-generator.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        run: |
+            errCount=0
+
+            echo "Files changes: ${{steps.filesChange.outputs.changed}}"
+            echo "Version bumped: ${{steps.versionBump.outputs.changed}}"
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:37]:
```suggestion
            echo "Bumped to version: ${{steps.versionBump.outputs.changed}}"
```

File: .github/workflows/version-generator.yml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: Detect version generator bump
+        id: versionBump
+        uses: ./.github/actions/detect-version-generator-bump
+
+      - name: Check version generator bump PR
+        run: |
+            errCount=0
+
+            echo "Files changes: ${{steps.filesChange.outputs.changed}}"
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:36]:
```suggestion
            echo "Changed files: ${{steps.filesChange.outputs.changed}}"
```

File: .github/workflows/version-generator.yml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+            echo "Version bumped: ${{steps.versionBump.outputs.changed}}"
+
+            if [[ "${{steps.filesChange.outputs.changed}}" == "false" && "${{steps.versionBump.outputs.changed}}" == "false" ]]
+            then
+              echo "::message::Generator not changed"
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:41]:
```suggestion
              echo "::message::The generator has no changes"
```

File: .github/workflows/version-generator.yml [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+              exit $errCount
+            fi
+
+            if [[ "${{steps.filesChange.outputs.changed}}" == "true" && "${{steps.versionBump.outputs.changed}}" == "false" ]]
+            then
+              echo "::error::Generator changed, but version is not bumped"
```
Comment (AnnaKondratova) [inline .github/workflows/version-generator.yml:47]:
```suggestion
              echo "::error:: The generator files are changed, but the version is not bumped"
```

File: .github/workflows/version-generator.yml (hunk 6 above)
Comment (OlegKipchatov) [reply .github/workflows/version-generator.yml:10]: It's default GA type
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Review by AnnaKondratova — DISMISSED** [review]

File: .github/actions/detect-generator-changes/action.yml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  steps:
+    - shell: bash
+      id: detection
+      run: |
+        diff=$( git diff HEAD^ --name-only -- packages/devextreme-angular-generator )
+
+        if [ "$diff" == "" ]
+        then
+          echo "::set-output name=changed::false"
+          echo "Generator files have no changes"
```
Comment (AnnaKondratova) [inline .github/actions/detect-generator-changes/action.yml:17]:
```suggestion
          echo "The generator files have no changes"
```

File: .github/workflows/version-generator.yml [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Detect changes in the generator files
+        id: filesChange
+        uses: ./.github/actions/detect-generator-changes
+
+      - name: Detect the generator version bump
```
Comment (IlyaKhD) [inline .github/workflows/version-generator.yml:28]:
```suggestion
      - name: Detect the generator version bump
        if: ${{steps.filesChange.outputs.changed}}
```

**Pull request closed without merging**